base/setup/reactos/lang/en-US.rc

```diff
@@ -173,3 +176,29 @@ BEGIN
     IDS_PARTITION_SIZE "Size"
     IDS_PARTITION_STATUS "Status"
 END
+
+STRINGTABLE
+BEGIN
+    IDS_BOOTLOADER_NOINST    "No installation"
+    IDS_BOOTLOADER_REMOVABLE "Removable media"
+    IDS_BOOTLOADER_SYSTEM    "System partition (Default)" // For non-MBR disks
+    IDS_BOOTLOADER_MBRVBR    "MBR and VBR (Default)"      // For MBR disks only
+    IDS_BOOTLOADER_VBRONLY   "VBR only"                   // ""
+END
+
+/* Error Strings */
+
+STRINGTABLE
+BEGIN
+    IDS_ERROR_INVALID_INSTALLDIR_CHAR_TITLE "Invalid character"
+    IDS_ERROR_INVALID_INSTALLDIR_CHAR "The only valid characters are:\n\
+Alpha-numeric (a-z, A-Z, 0-9), and\n . \\ - _\n\
+White-space is not allowed."
+
+// ERROR_DIRECTORY_NAME
+    IDS_ERROR_DIRECTORY_NAME_TITLE "Invalid installation directory"
+    IDS_ERROR_DIRECTORY_NAME "The ReactOS installation directory must consist of valid 8.3 components, \
```

Comment:
What does this even mean from the perspective of the user?
"The ReactOS installation directory must consist of valid 8.3 components"
The fact that the install directory consists of files that follow the 8.3 naming convention does not matter to the user. Nobody will understand what "8.3" means.
How about:
"The ReactOS installation directory can only contain letters, numbers, and periods. Spaces are not allowed."

Reply:
> What does this even mean from the perspective of the user?

Well, since in this case the user installs an _operating system_ (and not just a mere game), this user should be computer-literate a minima.

>  The fact that the install directory consists of files that follow the 8.3 naming convention does not matter to the user.

It's not the directory that "consists of files" that follow 8.3, it's each of the name _**components**_ in the path. (So, in the case of an installation path: `\ReactOS\old_ver\0_4.0`, each piece: `ReactOS`, `old_ver`, and `0_4.0` needs to conform to 8.3 name.)

Why? For backward compatibility with old programs that want to access the SystemRoot.

----

In effect, each component can contain letters, numbers, _**some**_ type of periods (not all), some other characters (but not all), and no spaces.
In the case of Windows installer, it says something a bit more vague (so as the user you may need to guess what is allowed and what is not):
![image](https://github.com/user-attachments/assets/89e8f717-003f-468b-8634-afea1c27ed0b)
Translation:
> The directory name you have entered is not valid. Please verify that the directory name is not the root directory and does not contain two consecutive backslash characters.
> Please also verify that the directory name respects the standard naming rules of MS-DOS file names.
> Press ENTER to continue. The installation program will ask you to enter a different directory name.

As you can see, in the example above, the "8.3" format name and the allowed/forbidden characters is hidden behind the more vague `respects the standard naming rules of MS-DOS file names` expression.

On the other hand, and a bit similar to what I had in mind, Windows is a bit more explicit elsewhre, for example in the shell:
![error_balloons_for_edit_boxes](https://github.com/user-attachments/assets/a20c7201-56e2-41bd-8a5c-ebfbfd5979f1)

Reply:
I do think what @cbialorucki wanted to express with "What does this even mean from the perspective of the user?" is that the components-aspect (all files and all subfolders within the Windows directory) is not what the user does actually specify here. He most of the time will just decide on the topmost folders name. So the suggestion from @binarymaster above is good imho to express that limitation accurately enough, and takes a bit of mental load from the users shoulders by avoiding the "concept of components within the path". Microsofts vague way to express theirselve looks clever in the meaning that they most likely tried to avoid having to refine that text a thousand times.

Reply:
> the components-aspect (all files and all subfolders within the Windows directory) is not what the user does actually specify here.

By "path component", I mean, each of the names that are separated with path separators `\`, for example:
`\ReactOS\new\0_4.15` , the path components are: `ReactOS`, `new` and `0_4.15`.
(same term as in https://learn.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-pathfindnextcomponentw )

Reply:
Suggestion addressed: see latest commit (en-US.rc file)